src/utilities/idf/WorkspaceObject.cpp

```diff
@@ -359,7 +359,7 @@ namespace detail {
     if (checkValidity && (level > StrictnessLevel::None)) {
 
       // do not set if would violate field NullAndRequired
-      if ((level > StrictnessLevel::Draft) && newName.empty() && iddObject().isRequiredField(*index)) {
+      if ((level >= StrictnessLevel::Draft) && newName.empty() && iddObject().isRequiredField(*index)) {
```

Comment:
I thought this was enforced at `Draft` strictness, we don't use `Final` strictness for anything that I know of